### ISSUE TITLE
get_cmap error when using icenet_plot_forecast

### ISSUE THREAD

**Issue by EllieBowler** [icenet-ai/icenet#277]
* IceNet version: 0.2.8
* Python version: 3.11
* Operating System: Linux - BAS HPC

### Description

Running icenet_plot_forecast and get error that matplotlib.cm has no attribute get_cmap. 
For env set up followed instructions from [icenet-notebook](https://github.com/icenet-ai/icenet-notebooks/blob/main/01.cli_demonstration.ipynb) - created fresh conda env and pip installed icenet. @bnubald 

### What I Did
```
icenet_plot_forecast north predict/carib_2021_north.nc 2021-10-01 -o outputs -r 10,20,20,30
```
```
2024-06-14 13:43:40.176424: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-06-14 13:43:40.217431: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-06-14 13:43:40.217574: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-06-14 13:43:40.218571: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-06-14 13:43:40.225369: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-06-14 13:43:46.680289: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
PyTorch not found - not required if not using PyTorch
usage: icenet_plot_forecast [-h] [-o OUTPUT_PATH] [-v] [-r REGION] [-l LEADTIMES] [-c] [-f {mp4,png,svg,tiff}] [-n CMAP_NAME] [-s] {north,south} forecast_file forecast_date
icenet_plot_forecast: error: argument hemisphere: invalid choice: 'predict/carib_2021_north.nc' (choose from 'north', 'south')
(/data/hpcdata/users/eller/conda-envs/icenet-wwf) 80 eller@bslws05>icenet_plot_forecast north predict/carib_2021_north.nc 2021-10-01 -o outputs -r 10, 20, 20, 30
2024-06-14 13:45:58.937163: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-06-14 13:45:58.979141: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-06-14 13:45:58.979206: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-06-14 13:45:58.980358: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-06-14 13:45:58.987853: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-06-14 13:46:00.290634: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
PyTorch not found - not required if not using PyTorch
usage: icenet_plot_forecast [-h] [-o OUTPUT_PATH] [-v] [-r REGION] [-l LEADTIMES] [-c] [-f {mp4,png,svg,tiff}] [-n CMAP_NAME] [-s] {north,south} forecast_file forecast_date
icenet_plot_forecast: error: argument -r/--region: invalid region_arg value: '10,'
(/data/hpcdata/users/eller/conda-envs/icenet-wwf) 81 eller@bslws05>icenet_plot_forecast north predict/carib_2021_north.nc 2021-10-01 -o outputs -r 10,20,20,30
2024-06-14 13:46:59.778733: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-06-14 13:46:59.819437: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-06-14 13:46:59.819486: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-06-14 13:46:59.820507: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-06-14 13:46:59.827388: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-06-14 13:47:00.977158: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
PyTorch not found - not required if not using PyTorch
WARNING:root:No directory at: outputs
INFO:root:Using cmap Blues_r
Traceback (most recent call last):
  File "/data/hpcdata/users/eller/conda-envs/icenet-wwf/bin/icenet_plot_forecast", line 8, in <module>
    sys.exit(plot_forecast())
             ^^^^^^^^^^^^^^^
  File "/data/hpcdata/users/eller/conda-envs/icenet-wwf/lib/python3.11/site-packages/icenet/plotting/forecast.py", line 1602, in plot_forecast
    cmap = cm.get_cmap(cmap_name)
           ^^^^^^^^^^^
AttributeError: module 'matplotlib.cm' has no attribute 'get_cmap'
```

**Comment by bnubald** [comment]
Hi @EllieBowler thanks for posting this.

Could you let me know the matplotlib version that was installed please?

**_Personal note:_** Looks like its caused by an API change in latest release of matplotlib last month, refer to change [here](https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.9.0.html#removals)

**Comment by bnubald** [comment]
I suspect the following will solve for now:

`pip install "matplotlib<3.9.0"`

This issue will affect new installs of older versions of IceNet as well where matplotlib>=3.9.0 is installed.

**Comment by EllieBowler** [comment]
Thanks Bryn! Pip installing an earlier matplotlib as above solved the issue (downgraded from 3.9.0 to 3.8.4)
